internal/controller/grafanaorganization_controller.go

```diff
@@ -39,24 +44,91 @@ type GrafanaOrganizationReconciler struct {
 
 // Reconcile is part of the main Kubernetes reconciliation loop which aims to
 // move the current state of the GrafanaOrganization closer to the desired state.
-// TODO(zirko): Modify the Reconcile function to compare the state specified by
-// the GrafanaOrganization object against the actual organization state, and then
-// perform operations to make the organization state reflect the state specified by
-// the user.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.0/pkg/reconcile
 func (r *GrafanaOrganizationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
 
-	// TODO(zirko): your logic here
+	logger.Info("Started reconciling Grafana Organization")
+	defer logger.Info("Finished reconciling Grafana Organization")
+
+	grafanaOrganization := &v1alpha1.GrafanaOrganization{}
+	err := r.Client.Get(ctx, req.NamespacedName, grafanaOrganization)
+	if err != nil {
+		return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+	}
+
+	logger.WithValues("grafanaOrganization", grafanaOrganization.ObjectMeta.Name)
+
+	// Generate Grafana client
+	grafanaAPI, err := grafanaClient.GenerateGrafanaClient(ctx, r.Client, logger)
+	if err != nil {
+		logger.Error(err, "Failed to create Grafana admin client")
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	// Test connection to Grafana
+	_, err = grafanaAPI.Health.GetHealth()
```

Comment:
I was wondering if this is really needed, as we are going to send a request to grafana anyway, so if it's not healthy we would get an error. It feels like we are adding an additional request to grafana, but I'm not sure it's bringing much value.

Reply:
We will get rid of it in the next PRs :D This is the first PR of many and we wanted to test the client creation worked :)

Reply:
This is already removed  in the following PR :)